source/MaterialXTest/MaterialXCore/Document.cpp

```diff
@@ -116,3 +119,193 @@ TEST_CASE("Document", "[document]")
     // Validate the combined document.
     REQUIRE(doc->validate());
 }
+
+void printDifferences(mx::ElementEquivalenceResult& result, const std::string& label)
+{
+    size_t differenceCount = result.differenceCount();
+    for (size_t i=0; i<differenceCount; ++i)
+    {
+        mx::StringVec message = result.getDifference(i);
+        std::cout << label << ": " << "Element: " << message[0] << 
+            ", Element: " << message[1] << ", Difference Type: " << message[2] 
+            << ", Value: " << message[3] << std::endl;
+    }
+}
+
+TEST_CASE("Document equivalence", "[document]")
```

Comment:
Looking at the current build errors, they seem to be platform-specific differences in the behavior of this new unit test!